### PR TITLE
Make decodeBnsAddress more strict and better tested

### PR DESCRIPTION
**Pull request by webmaster128**

### CODE REVIEW

File: packages/iov-bns/src/util.ts [hunk 1]
```diff
@@ -38,8 +38,17 @@ export function encodeBnsAddress(prefix: "iov" | "tiov", bytes: Uint8Array): Add
 }
 
 /** Decodes a printable address into bech32 object */
-export function decodeBnsAddress(address: Address): { readonly prefix: string; readonly data: Uint8Array } {
-  return Bech32.decode(address);
+export function decodeBnsAddress(
+  address: Address,
+): { readonly prefix: "iov" | "tiov"; readonly data: Uint8Array } {
```
Comment (willclarktech) [inline packages/iov-bns/src/util.ts]: Is it time for an `IovBech32Prefix` type?

File: packages/iov-bns/src/util.spec.ts [hunk 2]
```diff
@@ -27,6 +27,41 @@ describe("Util", () => {
     });
   });
 
+  describe("decodeBnsAddress", () => {
+    it("works for simple testnet address", () => {
+      expect(decodeBnsAddress("tiov17m9dug55pryn525drqwk9m7wgmlkp5ss4j2mky" as Address)).toEqual({
```
Comment (willclarktech) [inline packages/iov-bns/src/util.spec.ts]: Generation comment?

Comment (webmaster128) [reply packages/iov-bns/src/util.spec.ts]: Added

File: packages/iov-bns/src/util.ts (hunk 1 above)
Comment (webmaster128) [reply packages/iov-bns/src/util.ts]: Maybe, I don't know. Not too urgent in my opinion.

Comment (willclarktech) [reply packages/iov-bns/src/util.ts]: Not urgent but we awkwardly repeat `"iov" | "tiov"` in a lot of places.

Comment (webmaster128) [reply packages/iov-bns/src/util.ts]: okay okay, you get it :)

File: packages/iov-bns/src/util.ts [hunk 3]
```diff
@@ -28,19 +28,21 @@ import { As } from "type-tagger";
 
 import * as constants from "./constants";
 
-export function addressPrefix(chainId: ChainId): "iov" | "tiov" {
+export type IovBech32Prefix = "iov" | "tiov";
```
Comment (willclarktech) [inline packages/iov-bns/src/util.ts]: 😵

**Pull request merged**